tests/default_config.yml

```diff
@@ -244,6 +244,7 @@ namelist:
     swhtr: true
     trans_trac: true
     use_ufo: true
+    override_surface_radiative_fluxes: true
```

Comment:
Do we want this in the default test? If we do want to test this behavior, is this sufficient? Either here or in fv3net, you should add a test which involves setting override values and confirming they get used.

Reply:
What do you perceive is the harm of including them in the default test?  We would need to add special logic to skip testing these particular getters/setters in the wrapper test suite if we did not use this namelist parameter (since memory would not be allocated for the fields in the fortran model, and the wrapper tests by default test all fields defined in the JSON files).

I think testing the override capability a little more rigorously is a good idea, and the wrapper repo might be a good place to do this.  The main things I would check are:
1. The values end up accurately reflected in the fortran radiative flux diagnostics.  (Since the wrapper provides a getter for these buckets, this might be reasonably clean to set up).
2. The model produces different results than when run in default mode.  (I could envision setting the fluxes in some way, while using the same configuration as a regression test and checking that the checksums of the results don't match).

Would you be OK if I added tests to that effect here?

Reply:
> What do you perceive is the harm of including them in the default test? We would need to add special logic to skip testing these particular getters/setters in the wrapper test suite if we did not use this namelist parameter (since memory would not be allocated for the fields in the fortran model, and the wrapper tests by default test all fields defined in the JSON files).

Ah, I didn't realize that the getters and setters fail (probably segfault) for these variables if this namelist variable is False. Any segfault is a bug. When that's the case, it's probably more reasonable to raise the exception you get when a variable is not one of the ones available. An alternative "fix" is to allocate memory for these variables in the Fortran regardless of whether the data is used. Please implement one of these (or something else you can think of), since otherwise this behavior will be very hard to document when we make a proper list of available variables.

The tests you suggested sound good. I would add that for test (2), if you're using checksums then the checksums should match their previous saved value. Otherwise you might have a case where this test "passes" simply because the bit-for-bit behavior of the model was changed. If you're able to use something like `not np.allclose(value, target)` where target is something you can analytically determine (maybe if you set the fluxes to zero, this will be easy), then simply checking that e.g. the fluxes are nonzero or zero based on the namelist flag is enough.

Reply:
> When the data isn't initialized, it's probably more reasonable to raise the exception you get when a variable is not one of the ones available.

Is there a way to check this from Python?

> An alternative "fix" is to allocate memory for these variables in the Fortran regardless of whether the data is used. 

I am reluctant to do this, because I want to keep the default fortran model configuration as unchanged as possible for the DSL team, but we can probably consider it if alternatives are difficult.

> If you're able to use something like `not np.allclose(value, target)` where target is something you can analytically determine (maybe if you set the fluxes to zero, this will be easy), then simply checking that e.g. the fluxes are nonzero or zero based on the namelist flag is enough.

The existing getter/setter tests would check that the state of the payload variables is actually set (which is important); checking the radiative flux diagnostics are updated properly can be done analytically, but that's not sufficient to say that the fluxes were actually used. Checking that the radiative fluxes got passed to the land surface model really can only be checked by seeing if the model results -- i.e. more than just the radiative flux diagnostics -- change (I agree though locking in the checksums would be useful).

Reply:
> Is there a way to check this from Python?

You'd probably want to make a getter for the namelist variable.

> Checking that the radiative fluxes got passed to the land surface model really can only be checked by seeing if the model results -- i.e. more than just the radiative flux diagnostics -- change

I think it would be great to check that as well.

Reply:
> > When the data isn't initialized, it's probably more reasonable to raise the exception you get when a variable is not one of the ones available.
> 
> Is there a way to check this from Python?

I'm working on a way to do this through adding a way to get the `gfs_physics_nml` namelist flags (following the `flagstruct_properties` code closely).

Reply:
> You'd probably want to make a getter for the namelist variable.

Awesome, it looks we're on the same page.  I'm adding tests for the flags as well (it looks like they were not tested before?).

Reply:
> I'm adding tests for the flags as well

That's great, thanks!

Reply:
Cool, I added the flags and fortran radiative flux diagnostics tests.  I think the checksum tests may be easier to add in the fortran repo.  I'm thinking of just adding a config that is identical to the default with the exception that it sets the `override_surface_radiation_fluxes` flag to `.true.`; by default the overrides are set to zero, which is certainly different than what they will be by default, so the model should produce different results.

Reply:
Actually I have an alternative idea, which I think may be easier.

Reply:
I implemented my idea here: https://github.com/VulcanClimateModeling/fv3gfs-wrapper/pull/244/commits/56657a5ff8902f4c35498be29e614db3c7c64892.  Pretty cool that we can run the model with this much control.  I verified that running an identical step twice with this approach indeed produces an identical result.